src/components/MarkdownLayout/TableOfContents/getGithubUrl.tsx

```diff
@@ -4,18 +4,35 @@ import { graphql, useStaticQuery } from 'gatsby';
 import { SolutionInfo } from '../../../models/solution';
 import { ModuleInfo } from '../../../models/module';
 
-export default function getGithubUrl() {
+export default function getInfoToEdit() {
   const moduleLayoutInfo = useContext(MarkdownLayoutContext);
   const markdownInfo = moduleLayoutInfo.markdownLayoutInfo;
   const relativePath = markdownInfo.fileRelativePath;
-  let githubUrl = 'https://github.com/cpinitiative/usaco-guide/blob/master/';
-
+  const githubUrlPrefix =
+    'https://github.com/cpinitiative/usaco-guide/blob/master/';
+  const rawGithubUrlPrefix =
+    'https://raw.githubusercontent.com/cpinitiative/usaco-guide/master/';
+  let suffix = '';
   if (markdownInfo instanceof ModuleInfo) {
     // /\d_.*/.test(relativePath)
-    githubUrl += 'content/' + relativePath;
+    suffix = 'content/' + relativePath;
   } else {
-    githubUrl += 'solutions/' + relativePath;
+    suffix = 'solutions/' + relativePath;
   }
-  githubUrl = githubUrl.replace(' ', '%20'); // is there some function that can automatically escape chars?
+
+  const escapeSpaces = url => url.replace(' ', '%20');
```

Comment:
I think this is what you're looking for: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Reply:
Yeah, `encodeURI` works.